addons/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/test/java/org/nuxeo/ecm/core/bulk/TestSetContentLengthAction.java

```diff
@@ -0,0 +1,90 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.ecm.core.bulk;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.nuxeo.ecm.core.bulk.S3SetContentLengthAction.ACTION_NAME;
+import static org.nuxeo.ecm.core.bulk.message.BulkStatus.State.COMPLETED;
+
+import java.time.Duration;
+
+import javax.inject.Inject;
+
+import org.apache.logging.log4j.Logger;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.blob.binary.BinaryBlob;
+import org.nuxeo.ecm.core.bulk.computation.BulkScrollerComputation;
+import org.nuxeo.ecm.core.bulk.message.BulkCommand.Builder;
+import org.nuxeo.ecm.core.bulk.message.BulkStatus;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+@RunWith(FeaturesRunner.class)
+@Features(CoreFeature.class)
+@Deploy("org.nuxeo.ecm.core.test.tests:OSGI-INF/test-repo-core-types-contrib.xml")
+@Deploy("org.nuxeo.ecm.core.storage.binarymanager.s3")
+@RepositoryConfig(init = BlobDocumentSetRepositoryInit.class) public class TestSetContentLengthAction {
+
+    private static final Logger log = org.apache.logging.log4j.LogManager.getLogger(BulkScrollerComputation.class);
+
+    @Inject public BulkService service;
+
+    @Inject public CoreSession session;
+
+    @Inject public TransactionalFeature txFeature;
+
+    @Test public void testSetContentLength() throws Exception {
+        // Note that content length is updated only on S3BinaryManager
+        String nxql = "SELECT * from File";
+        dumpDocs("BEFORE", nxql);
+        String commandId = service.submit(new Builder(ACTION_NAME, nxql).repository(session.getRepositoryName())
+                                                                        .user("Administrator")
+                                                                        .param("force", "true")
+                                                                        .build());
+        assertTrue("Bulk action didn't finish", service.await(Duration.ofSeconds(60)));
+        BulkStatus status = service.getStatus(commandId);
+        log.info(status);
+        assertNotNull(status);
+        assertEquals(COMPLETED, status.getState());
+        txFeature.nextTransaction();
+        dumpDocs("AFTER", nxql);
```

Comment:
Can't we/shoudln't we check that the effective blob length has been updated on the documents' blob?

Reply:
no, because the action is only working on an s3 binary store, so nothing is going to be updated during the unit test (cf comment at the beginning) still it covers the non-s3 code part.